File: dec1/one.mjs
import fs from "fs/promises";

(async function () {
    let file = await fs.readFile("./input.txt", "utf-8");
    let one = [];
    let two = [];
    file.split("\n").forEach(line => {
        let nums = line.split(" ");
        one.push(+nums[0]);
        two.push(+nums[3]);
    });
    one.sort((a, b) => a - b);
    two.sort((a, b) => a - b);
    one.shift();
    let res = 0;
    for(let i = one.length; i--;){
        res += Math.abs(one[i] - two[i]);
    }
    console.log(res);
})()

File: dec1/one.rs
use std::fs;

fn main () {
    let file = fs::read_to_string("input.txt").unwrap();
    let mut one = Vec::new();
    let mut two = Vec::new();
    for line in file.lines() {
        let arr: Vec<&str> = line.split_whitespace().collect();
        one.push(arr[0].parse::<i32>().unwrap());
        two.push(arr[1].parse::<i32>().unwrap());
    }
    one.sort();
    two.sort();
    let mut sum = 0;
    for i in 0..one.len() {
        sum += (one[i] - two[i]).abs();
    }
    println!("{}", sum);
}